okhttp/third_party/okhttp/java/io/grpc/okhttp/internal/Platform.java

```diff
@@ -105,47 +119,71 @@ public void connectSocket(Socket socket, InetSocketAddress address,
   }
 
   /** Attempt to match the host runtime to a capable Platform implementation. */
+  @SuppressWarnings("unchecked")
```

Comment:
I've tried to keep the changes to a minimum but it's hard to avoid with the switch to detecting based on Provider rather than OpenSSLSocketImpl availability.

@swankjesse 
@JakeWharton 
@jhump 

Hey OkHTTP folks, any opinions on the changes here and potential negative impacts on Android. Could this change be upstreamed into OkHTTP?